Lib/ufo2ft/filters/flattenComponents.py

```diff
@@ -0,0 +1,53 @@
+from fontTools.misc.transform import Transform
+from ufo2ft.filters import BaseFilter
+
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class FlattenComponentsFilter(BaseFilter):
+
+    def __call__(self, font, glyphSet=None):
+        if super(FlattenComponentsFilter, self).__call__(font, glyphSet):
+            modified = self.context.modified
+            if modified:
+                logger.info('Flattened composite glyphs: %i' %
+                         len(modified))
+            return modified
+
+    def filter(self, glyph):
+        flattened = False
+        if not glyph.components:
+            return flattened
+        for comp in list(glyph.components):
+            flattened_tuples = _flattenComponent(comp)
+            if flattened_tuples[0] != (comp.baseGlyph, comp.transformation):
+                flattened = True
+            glyph.removeComponent(comp)
+            for flattened_tuple in flattened_tuples:
+                pen = glyph.getPen()
```

Comment:
you can get the pen outside the for loops, the glyph doesn't change so neither does the pen

Reply:
Good point.